### PR TITLE
[MFTF coverage] Adding MFTF coverage for saving the image preview with custom name

### PR DESCRIPTION
**Pull request by eduard13**
### Description (*)
This PR covers the saving image preview with a custom name story by MFTF

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#445: [MFTF] User sets the image name

### Manual testing scenarios (*)
N/A

### CODE REVIEW

File: AdobeStockImageAdminUi/Test/Mftf/ActionGroup/AdminSaveAdobeStockImagePreviewActionGroup.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (filmaj) [inline AdobeStockImageAdminUi/Test/Mftf/ActionGroup/AdminSaveAdobeStockImagePreviewActionGroup.xml:9]: There looks to be a similar ActionGroup already called `AdminAdobeStockSavePreviewActionGroup`. Are they similar enough that  we can consolidate these two ActionGroups in this PR?

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminSavesTheImagePreviewWithNewNameTest.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminSavesTheImagePreviewWithNewNameTest">
+        <annotations>
+            <features value="AdobeStockConfiguration"/>
+            <stories value="Admin should be able to save the image preview with a new image name"/>
+            <title value="Admin is able to set the image name on saving the image preview"/>
+            <description value="Admin is able to set the image name on saving the image preview"/>
+            <severity value="CRITICAL"/>
+            <group value="adobe_stock_integration_general"/>
```
Comment (filmaj) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminSavesTheImagePreviewWithNewNameTest.xml]: I think we should reuse the groups that other tests already have:

```suggestion
            <group value="adobe_stock_integration_filters"/>
            <group value="adobe_stock_integration"/>
```

**Review by filmaj — CHANGES_REQUESTED** [review]
I left one suggestion on a change and an open question regarding consolidating an action group. Other than that, I was able to run this test successfully locally on my instance. Nice work!

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminSavesTheImagePreviewWithNewNameTest.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        </annotations>
+        <before>
+            <actionGroup ref="LoginActionGroup" stepKey="login"/>
+            <actionGroup ref="AdminAdobeStockSetConfigActionGroup" stepKey="setCorrectModuleConfig"/>
```
Comment (filmaj) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminSavesTheImagePreviewWithNewNameTest.xml]: We can remove these two `before` `actionGroup`s as long as you accept my suggested change above for the tweaked group name - as that would then include the test in the main Suite we have defined (https://github.com/magento/adobe-stock-integration/blob/develop/AdobeStockImageAdminUi/Test/Mftf/Suite/adobe-stock-integration.xml#L12-L15) which runs these two tasks once before executing the tests comprising the suite.

Comment (filmaj) [reply AdobeStockImageAdminUi/Test/Mftf/Test/AdminSavesTheImagePreviewWithNewNameTest.xml]: Or maybe  not.. other tests do a similar thing.. perhaps worth revisiting later once MFTF test development settles down. Feel free to ignore my above comment for now.

Comment (sivaschenko) [reply AdobeStockImageAdminUi/Test/Mftf/Test/AdminSavesTheImagePreviewWithNewNameTest.xml]: That action group is called before the test suite anyway, there is no need to call it before the test.

**Comment by diazwatson** [comment]
Running this test in my local env I can see it successfully complete.
```
Magento\FunctionalTestingFramework.functional Tests (1) ------------------------
Modules: \Magento\FunctionalTestingFramework\Module\MagentoWebDriver, \Magento\FunctionalTestingFramework\Helper\Acceptance, \Magento\FunctionalTestingFramework\Helper\MagentoFakerData, \Magento\FunctionalTestingFramework\Module\MagentoSequence, \Magento\FunctionalTestingFramework\Module\MagentoAssert, Asserts
--------------------------------------------------------------------------------
AdminSavesTheImagePreviewWithNewNameTestCest: Admin saves the image preview with new name test
Signature: Magento\AcceptanceTest\_default\Backend\AdminSavesTheImagePreviewWithNewNameTestCest:AdminSavesTheImagePreviewWithNewNameTest
Test: tests/functional/Magento/FunctionalTest/_generated/default/AdminSavesTheImagePreviewWithNewNameTestCest.php:AdminSavesTheImagePreviewWithNewNameTest
Scenario --
[login] LoginActionGroup
  [navigateToAdmin] am on page "admin"
  [fillUsername] fill field "#username","admin"
  [fillPassword] fill field "#login","123123q"
  [click] click ".actions .action-primary"
  [clickWaitForPageLoad] wait for page load 30
[setCorrectModuleConfig] AdminAdobeStockSetConfigActionGroup
  [navigateToSystemConfigurationPage] am on page "/admin/admin/system_config/edit/section/system"
  [waitForPageLoad] wait for page load 10
  [scrollToAdobeStockIntegrationSection] scroll to "#system_adobe_stock_integration-head"
  [expandAdobeStockIntegrationTab] conditional click "#system_adobe_stock_integration-head","[data-ui-id='select-groups-adobe-stock-integration-fields-enabled-value']",false
  [enableOrDisableAdobeStock] select option "[data-ui-id='select-groups-adobe-stock-integration-fields-enabled-value']","1"
  [enterAdobeStockApiKey] fill field "[data-ui-id='password-groups-adobe-stock-integration-fields-api-key-value']",false
  [enterAdobeStockPrivateKey] fill field "[data-ui-id='password-groups-adobe-stock-integration-fields-private-key-value']",false
  [saveConfiguration] click "#save"
  [waitForConfigurationToSave] wait for page load 10
[openMediaGalleryForPage] AdminOpenMediaGalleryForPageNoEditorActionGroup
  [amOnPagePagesGrid] am on page "/admin/cms/page"
  [waitForPageLoad] wait for page load 10
  [clickAddNewPage] click "#add"
  [clickAddNewPageWaitForPageLoad] wait for page load 30
  [clickExpandContent] click "div[data-index=content]"
  [waitForInsertImageButton] wait for element visible ".scalable.action-add-image.plugin",10
  [clickInsertImage] click ".scalable.action-add-image.plugin"
  [] wait for initial media gallery load, where the gallery chrome loads (and triggers loading modal)
  [waitForMediaGalleryInitialLoad] wait for page load 10
  [] javascript ERROR(SEVERE) - https://adobestock.test/static/version1572277603/adminhtml/Magento/backend/en_US/mage/adminhtml/browser.js 91:47 Uncaught T
  [] wait for second media gallery load, where the gallery images load (and triggers loading modal once more)
  [waitForMediaGallerySecondaryLoad] wait for page load 10
[openAdobeStockPopup] AdminAdobeStockOpenPanelFromMediaGalleryActionGroup
  [openAdobeStockPanel] click "[data-ui-id=wysiwyg-images-content-search-adobe-stock-button]"
  [waitForAdobeStockPanelOpen] wait for page load 10
[expandsFirstImageInGrid] AdminAdobeStockExpandImagePreviewActionGroup
  [clickOnThumbnail] click "aside.adobe-stock-modal [data-role=grid-wrapper] img[data-role=thumbnail]"
  [waitForImagePreviewToExpand] wait for loading mask to disappear
[openTheSaveImagePreviewPopup] AdminAdobeStockSavePreviewActionGroup
  [clickOnSavePreview] click "//button[@class='action-secondary']/span[contains(.,'Save Preview')]"
  [waitForPageLoad] wait for page load 10
[saveImagePreview] AdminSaveAdobeStockImagePreviewActionGroup
  [setImageName] fill field ".adobe-stock-save-preview-prompt input.admin__control-text[name='name']","my-image-5dc36ad58737f"
  [confirmSavingThePreview] click "//*[@class='action-primary action-accept']"
  [waitForImagePreviewToSave] wait for page load 10
[checkIfTheSavedImagesIsAvailable] AssertAdminImageIsVisibleInMediaGalleryActionGroup
  [waitForGalleryLoad] wait for page load 10
  [seeSavedImage] see element "img[alt='my-image-5dc36ad58737f.jpeg']"
[removeSavedPreview] AdminDeleteSelectedAdobeStockImagePreviewActionGroup
  [clickDeleteMedia] click "button#delete_files"
  [waitForConfirmationModal] wait for element visible "//*[@class='action-primary action-accept']",10
  [clickConfirmButton] click "//*[@class='action-primary action-accept']"
  [waitForDeletionPageLoad] wait for page load 10
[logout] logout
  [amOnPage] am on page "/admin/admin/auth/logout/"
 PASSED

--------------------------------------------------------------------------------

Time: 3.14 minutes, Memory: 8.00MB

OK (1 test, 1 assertion)
```

File: AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSaveImagePreviewSection.xml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<sections xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:mftf:Page/etc/SectionObject.xsd">
+    <section name="AdobeStockSaveImagePreviewSection">
+        <element name="name" type="text" selector=".adobe-stock-save-preview-prompt input.admin__control-text[name='name']"/>
+        <element name="mediaImage" type="text" selector="img[alt='{{name}}.jpeg']" parameterized="true"/>
```
Comment (rogyar) [inline AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSaveImagePreviewSection.xml]: I believe we should use a more unique selector here. The webdriver will look for `img[alt='{{name}}.jpeg]` on the entire page and there's a chance to spot the same image in the grid or so some day.

**Review by rogyar — CHANGES_REQUESTED** [review]

File: AdobeStockImageAdminUi/Test/Mftf/ActionGroup/AdminSaveAdobeStockImagePreviewActionGroup.xml (hunk 1 above)
Comment (eduard13) [reply AdobeStockImageAdminUi/Test/Mftf/ActionGroup/AdminSaveAdobeStockImagePreviewActionGroup.xml:9]: Hi @filmaj , the existing `AdminAdobeStockSavePreviewActionGroup`, just clicks on the Save Preview button, I don't think that is the right name of such ActionGroup. However, refactoring the `AdminAdobeStockSavePreviewActionGroup`, means that we need to adjust the existing tests.
The only improvement I can see, is to just rename to something `AdminAdobeStockOpenSavePreviewModalActionGroup`. 
Please let me know your thoughts.

File: AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSaveImagePreviewSection.xml (hunk 4 above)
Comment (eduard13) [reply AdobeStockImageAdminUi/Test/Mftf/Section/AdobeStockSaveImagePreviewSection.xml]: Agree. Updated the selector.

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminSavesTheImagePreviewWithNewNameTest.xml (hunk 3 above)
Comment (eduard13) [reply AdobeStockImageAdminUi/Test/Mftf/Test/AdminSavesTheImagePreviewWithNewNameTest.xml]: That makes sense, removed the `AdminAdobeStockSetConfigActionGroup` call before the test.
Thank you.

**Review by rogyar — APPROVED** [review]

**Comment by eduard13** [comment]
Hi @filmaj, @sivaschenko, any updates on this one?

**Review by diazwatson — APPROVED** [review]

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**